From 9ace3c6f3759c073eda66dbd4fe7ad0173af56df Mon Sep 17 00:00:00 2001
Message-ID: <9ace3c6f3759c073eda66dbd4fe7ad0173af56df.1705743627.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1705743627.git.sharlatanus@gmail.com>
References: <cover.1705743627.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Tue, 16 Jan 2024 23:10:23 +0000
Subject: [PATCH 11/38] gnu: go-filippo-io-edwards25519: Move to (gnu packages
 golang-crypto).

* gnu/packages/golang.scm (go-filippo-io-edwards25519): Move from here ...
* gnu/packages/golang-crypto.scm: ... to here.

* gnu/packages/password-utils.scm: Add (gnu packages golang-crypto) to
used modules.

Change-Id: I11769b8b212bf4cbe98fe10181c0a2a479c599df
---
 gnu/packages/golang-crypto.scm  | 24 ++++++++++++++++++++++++
 gnu/packages/golang.scm         | 23 -----------------------
 gnu/packages/password-utils.scm |  1 +
 3 files changed, 25 insertions(+), 23 deletions(-)

diff --git a/gnu/packages/golang-crypto.scm b/gnu/packages/golang-crypto.scm
index 324e65c1ae..72771d40a2 100644
--- a/gnu/packages/golang-crypto.scm
+++ b/gnu/packages/golang-crypto.scm
@@ -2,6 +2,7 @@
 ;;; Copyright © 2018 Pierre Neidhardt <mail@ambrevar.xyz>
 ;;; Copyright © 2020 Leo Famulari <leo@famulari.name>
 ;;; Copyright © 2021 Raghav Gururajan <rg@raghavgururajan.name>
+;;; Copyright © 2022 Nicolas Graves <ngraves@ngraves.fr>
 ;;; Copyright © 2023 Clément Lassieur <clement@lassieur.org>
 ;;;
 ;;; This file is part of GNU Guix.
@@ -40,6 +41,29 @@ (define-module (gnu packages golang-crypto)
 ;;;
 ;;; Code:
 
+(define-public go-filippo-io-edwards25519
+  (package
+    (name "go-filippo-io-edwards25519")
+    (version "1.0.0")
+    (source
+     (origin
+       (method git-fetch)
+       (uri (git-reference
+             (url "https://github.com/FiloSottile/edwards25519")
+             (commit (string-append "v" version))))
+       (file-name (git-file-name name version))
+       (sha256
+        (base32 "01m8hpaj0cwp250f7b0din09cf8j6j5y631grx67qfhvfrmwr1zr"))))
+    (build-system go-build-system)
+    (arguments
+     '(#:import-path "filippo.io/edwards25519"))
+    (home-page "https://filippo.io/edwards25519")
+    (synopsis "Group logic for the twisted Edwards curve")
+    (description "This package implements the edwards25519 elliptic curve in
+Go, exposing the necessary APIs to build a wide array of higher-level
+primitives.")
+    (license license:bsd-3)))
+
 (define-public go-github-com-aead-chacha20
   (let ((commit "8b13a72661dae6e9e5dea04f344f0dc95ea29547")
         (revision "0"))
diff --git a/gnu/packages/golang.scm b/gnu/packages/golang.scm
index 41bc99b146..4e44ebf5ba 100644
--- a/gnu/packages/golang.scm
+++ b/gnu/packages/golang.scm
@@ -9792,29 +9792,6 @@ (define-public age-keygen
        #:unpack-path "filippo.io/age"
        #:install-source? #f))))
 
-(define-public go-filippo-io-edwards25519
-  (package
-    (name "go-filippo-io-edwards25519")
-    (version "1.0.0")
-    (source
-     (origin
-       (method git-fetch)
-       (uri (git-reference
-             (url "https://github.com/FiloSottile/edwards25519")
-             (commit (string-append "v" version))))
-       (file-name (git-file-name name version))
-       (sha256
-        (base32 "01m8hpaj0cwp250f7b0din09cf8j6j5y631grx67qfhvfrmwr1zr"))))
-    (build-system go-build-system)
-    (arguments
-     '(#:import-path "filippo.io/edwards25519"))
-    (home-page "https://filippo.io/edwards25519")
-    (synopsis "Group logic for the twisted Edwards curve")
-    (description
-     "This package implements the edwards25519 elliptic curve in Go, exposing
-the necessary APIs to build a wide array of higher-level primitives.")
-    (license license:bsd-3)))
-
 (define-public go-gitlab-com-yawning-edwards25519-extra
   (let ((commit "2149dcafc266f66d2487f45b156f6397f9c4760b")
         (revision "0"))
diff --git a/gnu/packages/password-utils.scm b/gnu/packages/password-utils.scm
index bcd323ced8..6dc257dc0e 100644
--- a/gnu/packages/password-utils.scm
+++ b/gnu/packages/password-utils.scm
@@ -91,6 +91,7 @@ (define-module (gnu packages password-utils)
   #:use-module (gnu packages gnome)
   #:use-module (gnu packages gnupg)
   #:use-module (gnu packages golang)
+  #:use-module (gnu packages golang-crypto)
   #:use-module (gnu packages gtk)
   #:use-module (gnu packages guile)
   #:use-module (gnu packages kerberos)
-- 
2.41.0

